ecc/p384/opt_test.go

```diff
@@ -4,14 +4,43 @@
 package p384
 
 import (
+	"bytes"
 	"crypto/elliptic"
+	"crypto/rand"
 	"math/big"
 	"testing"
 
 	"github.com/cloudflare/circl/internal/test"
 )
 
 func TestInternals(t *testing.T) {
+	t.Run("reduceScalar", func(t *testing.T) {
+		var c curve
+		var order = c.Params().N
+		var buffer [3 * sizeFp]byte
+
+		for i := 0; i < sizeFp; i++ {
+			buffer[i] = 0xFF
+		}
+		_, err := rand.Read(buffer[sizeFp:])
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for i := sizeFp; i < 3*sizeFp; i++ {
+			k := buffer[:i]
```

Comment:
Do we also want to test the case where `len(k) < sizeFp`? It looks like this just tests the cases where `len(k) >= sizeFp`.

Reply:
typo, fixing now.